### PR TITLE
Make delivery report mandatory for Scout uploads

### PR DESCRIPTION
**Pull request by ivadym**
## Description

Making the delivery report key mandatory for Scout uploads has required its inclusion in the constructor. This is because we dynamically build the `scout_load.yaml` file on the fly during uploads.

This kind of breaks the structure of including case and sample files and having the tags defined in one place. An alternative solution (a one-line code change) would have been to use an empty default string for the delivery report key as the default value. In this case, Scout would be responsible for failing the upload since it checks if the report file, if provided, exists.

### Changed
- Delivery report mandatory for Scout uploads
- Renamed function `get_report_accreditation` to `is_report_accredited`

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_cg -t cg -b [THIS-BRANCH-NAME] -a
    ```

### How to test

- [x] `cg upload -c <case_id>`

### Expected test outcome

- [ ] Delivery report generated for Cancer cases
- [x] Delivery report generated for Rnafusion cases
- [x] Upload fails when no delivery report has been found

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

File: cg/meta/upload/scout/balsamic_config_builder.py [hunk 1]
```diff
@@ -22,7 +23,9 @@ def __init__(self, hk_version_obj: Version, analysis_obj: Analysis, lims_api: Li
         )
         self.case_tags: CaseTags = CaseTags(**BALSAMIC_CASE_TAGS)
         self.sample_tags: SampleTags = SampleTags(**BALSAMIC_SAMPLE_TAGS)
-        self.load_config: BalsamicLoadConfig = BalsamicLoadConfig(track="cancer")
+        self.load_config: BalsamicLoadConfig = BalsamicLoadConfig(
+            track="cancer", delivery_report=self.get_file_from_hk({HK_DELIVERY_REPORT_TAG})
```
Comment (henrikstranneheim) [inline cg/meta/upload/scout/balsamic_config_builder.py:27]: Do we have any constants defined for the track?

File: cg/meta/upload/scout/mip_config_builder.py [hunk 2]
```diff
@@ -36,7 +33,9 @@ def __init__(
         )
         self.case_tags: CaseTags = CaseTags(**MIP_CASE_TAGS)
         self.sample_tags: SampleTags = SampleTags(**MIP_SAMPLE_TAGS)
-        self.load_config: MipLoadConfig = MipLoadConfig(track="rare")
+        self.load_config: MipLoadConfig = MipLoadConfig(
+            track="rare", delivery_report=self.get_file_from_hk({HK_DELIVERY_REPORT_TAG})
```
Comment (henrikstranneheim) [inline cg/meta/upload/scout/mip_config_builder.py:37]: rare what? Would be nice with a more descriptive name.

File: tests/apps/scout/test_scout_load_config.py [hunk 3]
```diff
@@ -1,4 +1,5 @@
 """Tests for the models in scout load config"""
```
Comment (henrikstranneheim) [inline tests/apps/scout/test_scout_load_config.py:1]:
```suggestion
"""Tests for the models in Scout load config."""
```

File: cg/meta/upload/scout/mip_config_builder.py (hunk 2 above)
Comment (ivadym) [reply cg/meta/upload/scout/mip_config_builder.py]: It's defined as `Literal["rare", "cancer"]` in Scout but I've created some constants for the tracks, good catch 👍

**Comment by ivadym** [comment]
Deployed

> Log deploy... done.
> cg, version 54.10.5

**Pull request merged**